### PR TITLE
Add summary task

### PR DESCRIPTION
**Pull request by Yun-Wu**

### CODE REVIEW

File: ecoscope_workflows/tasks/analysis/_aggregation.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+from typing import Annotated
```
Comment (cisaacstern) [inline ecoscope_workflows/tasks/analysis/_aggregation.py:1]:
```suggestion
from typing import Annotated, Literal
```
Goes with `func_name` type hint suggestion below.

File: ecoscope_workflows/tasks/analysis/_aggregation.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+from ecoscope_workflows.annotations import DataFrame, JsonSerializableDataFrameModel
+from ecoscope_workflows.decorators import distributed
+
+
+@distributed
+def aggregate(
+    df: DataFrame[JsonSerializableDataFrameModel],
+    column_name: Annotated[str, Field(description="Start date")],
+    func_name: str,
```
Comment (cisaacstern) [inline ecoscope_workflows/tasks/analysis/_aggregation.py:14]:
```suggestion
    func_name: Annotated[Literal["count", "mean", "sum"], Field(description="The method of aggregation".)
```
Perhaps we can constrain with a Literal here? So when called in `.replace(validate=True)` mode, Pydantic will throw a validation error if on of the specified literal values is not given.

File: ecoscope_workflows/tasks/analysis/_aggregation.py (hunk 1 above)
Comment (cisaacstern) [inline ecoscope_workflows/tasks/analysis/_aggregation.py:1]:
```suggestion
from typing import Annotated, Literal
```
Goes with func_name type hint suggestion.

File: ecoscope_workflows/tasks/analysis/_aggregation.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+from pydantic import Field
+
+from ecoscope_workflows.annotations import DataFrame, JsonSerializableDataFrameModel
+from ecoscope_workflows.decorators import distributed
+
+
+@distributed
+def aggregate(
+    df: DataFrame[JsonSerializableDataFrameModel],
+    column_name: Annotated[str, Field(description="Start date")],
```
Comment (cisaacstern) [inline ecoscope_workflows/tasks/analysis/_aggregation.py:13]: Guessing `"Start date"` is a typo?

File: ecoscope_workflows/tasks/analysis/_aggregation.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+    match func_name.upper():
+        case "COUNT":
+            return pd.Series({f"{column_name}_count": len(df)})
+        case "MEAN":
+            return pd.Series({f"{column_name}_mean": df[column_name].mean()})
+        case "SUM":
+            return pd.Series({f"{column_name}_sum": df[column_name].sum()})
+        case _:
+            raise ValueError(f"Unknown aggregation function: {func_name}")
+            raise ValueError(f"Unknown aggregation function: {func_name}")
```
Comment (cisaacstern) [inline ecoscope_workflows/tasks/analysis/_aggregation.py:25]:
```suggestion
            raise ValueError(f"Unknown aggregation function: {func_name}")
```

File: tests/tasks/test_aggregate.py [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import pandas as pd
+import pytest
+
+from ecoscope_workflows.tasks.analysis import (
+    aggregate,
+)  # Replace 'your_module' with the actual module name
```
Comment (cisaacstern) [inline tests/tasks/test_aggregate.py:6]:
```suggestion
)
```

File: tests/tasks/test_aggregate.py [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+def test_sum():
+    df = pd.DataFrame({"data": [1, 2, 3]})
+    result = aggregate(df, "data", "sum")
+    pd.testing.assert_series_equal(result, pd.Series({"data_sum": 6}))
+
+
+def test_invalid_function():
+    df = pd.DataFrame({"data": [1, 2, 3]})
+    with pytest.raises(ValueError):
+        aggregate(df, "data", "invalid_function")
```
Comment (cisaacstern) [inline tests/tasks/test_aggregate.py:30]: Thanks for always putting such nice readable unit tests. Very much appreciated!

**Review by cisaacstern — COMMENTED** [review]
Looking good @Yun-Wu! Just a few random nits.

**Comment by Yun-Wu** [comment]
> Looking good @Yun-Wu! Just a few random nits.

Thanks Charles! Somehow I forgot to push the latest commit, leaving the dirty code there. Thanks for catching them!

**Review by cisaacstern — APPROVED** [review]
Thanks @Yun-Wu!

One small point I noticed is that, while the return value is annotated as a `DataFrame[...]` it seems from the unit tests that the actual return type is a `Series` ... when we get into the `.replace(validate=True)` setting (e.g. a compiled script) I wonder if this will cause validation errors. 🤷 It's a small point though and I'm happy to see this go in now and we can handle that if/when it comes up in a separate ticket.

Will leave it to you to merge!

**Comment by Yun-Wu** [comment]
> Thanks @Yun-Wu!
> 
> One small point I noticed is that, while the return value is annotated as a `DataFrame[...]` it seems from the unit tests that the actual return type is a `Series` ... when we get into the `.replace(validate=True)` setting (e.g. a compiled script) I wonder if this will cause validation errors. 🤷 It's a small point though and I'm happy to see this go in now and we can handle that if/when it comes up in a separate ticket.
> 
> Will leave it to you to merge!

Thank you Charles. Created this ticket to follow up: https://github.com/wildlife-dynamics/ecoscope-workflows/issues/122

**Pull request merged**